### ISSUE TITLE
Does WebAssembly.compileStreaming allow a Response as parameter?

### ISSUE THREAD

**Issue by gahaas** [WebAssembly/spec#806]
The spec defines the signature of `WebAssembly.compileStreaming` as `Promise<Module> compileStreaming(Promise<Response> source);`. In my understanding this means that the parameter has to be a `Promise`, it cannot be a `Response` directly. The not below the signature of `WebAssembly.compileStreaming`, however, says 

> This algorithm accepts a Response object, or a promise for one ...

The design repo agrees with the note.

I think the spec and the note should match, so should the spec be updated or the note?

**Comment by lukewagner** [comment]
You're right that the intended behavior is to accept `Response` objects and I think this is handled automatically for us [by the definition of Web IDL `Promise` argument types](https://heycam.github.io/webidl/#es-promise) ([see also](https://www.w3.org/2001/tag/doc/promises-guide/#resolve-arguments)) which specify that a given JS argument value `v` is first `Promise.resolve`d.

**Comment by littledan** [comment]
Yes, seems like the note should be changed.

**Comment by littledan** [comment]
@gahaas Yes, I agree with your analysis. Basically, Promises mean that, across the JS universe, the method named "then" should only be used for Promise-related purposes.
